src/cc_bstring.c

```diff
@@ -142,3 +142,41 @@ bstring_atou64(uint64_t *u64, struct bstring *str)
 
     return CC_OK;
 }
+
+struct bstring *
+bstring_alloc(uint32_t size)
+{
+    struct bstring bs;
+    bstring_init(&bs);
+
+    bs.len = size;
+    bs.data = cc_alloc(size);
+    if (bs.data == NULL) {
+        return NULL;
+    }
+
+    struct bstring *rv = cc_alloc(sizeof(struct bstring));
```

Comment:
nit: prefer `sizeof(*rv)` in this case. it probably doesn't make much of a difference here, but by using `sizeof()` this way, if the type of `rv` changes, we can avoid making this change in multiple places.

Reply:
ah! good to know!